Assets/Scripts/UI/MintDeckButtonHandler.cs

```diff
@@ -0,0 +1,26 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.UI;
+
+[RequireComponent(typeof(Button))]
+public class MintDeckButtonHandler : MonoBehaviour
+{
+    void Start()
+    {
+        GetComponent<Button>().onClick.AddListener(OnButtonClicked);
+    }
+
+    async void OnButtonClicked()
+    {
+        GetComponent<Button>().interactable = false;
+        var tx = await PepemonCardDeck.CreateDeck();
+        // TODO: show error label on failure
+        if (string.IsNullOrEmpty(tx))
+        {
+            // TODO: enum screens
+            FindObjectOfType<MainMenuController>().ShowScreen(5);
+        }
+        GetComponent<Button>().interactable = true;
```

Comment:
Does `await` wait until the transaction is mined? If so, maybe we could refresh the UI at this point.

Reply:
That's a good idea. This PR does not have the deck loading implementation yet, but I can add it if that's what you meant by refreshing the UI.